matlab_simulation/05-estimation/fastSLAM/featureSearch_fs.m

```diff
@@ -0,0 +1,36 @@
+%% FUNC to search of features in Field of VIEW
+% INPUTS:(in order of call)
+% -------
+% No of features
+% Map of the Environment
+% Robot's pose at the current time step
+% Maximum Range
+% Maximum FOV
+% OUTPUTS:(in order of call)
+% --------
+% A vector containing all the measured indexs
+% % AUTHOR: BISMAYA SAHOO, EMAIL:bsahoo@uwaterloo.ca
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Feature Search
+    function meas_ind=featureSearch_fs(noFeatures,map,robot_pose,rmax,thmax)
+        meas_ind = [];
+        for i=1:noFeatures
+                if(inview(map(:,i),robot_pose,rmax,thmax))
+                    meas_ind=[meas_ind,i];%concatenate the indices
+                end
+        end
+            function yes = inview(f,x, rmax, thmax)
+            % Checks for features currently in robot's view
```

Comment:
"As long as it is working fine" is not a good measure for readable code. The fact that there is a global function `inview()` and you defining a local function **later** after `inview()` is first called confuses the reader, if the reader is not well versed in matlab's function precedence he/she will not know which version of `inview` will be called. People are lazy, they don't want to read the doc to find that out, neither do I. My suggestion is to either use the same `inview` function or atleast give it a different name.

Further more nested functions are generally not considered good practice, since if you have the justification for defining a function, it should be available for anyone to call, else it should not be a function. It is indeed a matter of taste, but I can assure you it is seldom used in general.